third_party/blink/renderer/core/dom/scripted_idle_task_controller.cc

```diff
@@ -37,26 +37,33 @@ class IdleRequestCallbackWrapper
       scoped_refptr<IdleRequestCallbackWrapper> callback_wrapper,
       base::TimeTicks deadline) {
 
-    recordreplay::Assert("[RUN-1335-1336] IdleTaskFired A %d",
-                          callback_wrapper->Id());
-
-    if (ScriptedIdleTaskController* controller =
-            callback_wrapper->Controller()) {
-      // If we are going to yield immediately, reschedule the callback for
-      // later.
-
-      recordreplay::Assert("[RUN-1335-1456] IdleTaskFired B %d",
-                           callback_wrapper->Id());
-
-      if (ThreadScheduler::Current()->ShouldYieldForHighPriorityWork()) {
-        controller->ScheduleCallback(std::move(callback_wrapper),
-                                     /* timeout_millis */ 0);
-        return;
+    // [RUN-1335] This might execute even after the controller has been marked
+    // for collection. We avoid possible divergence by checking for whether 
+    // it is still registered first.
+    bool hasCallback = callback_wrapper->Controller() &&
+                         callback_wrapper->Controller()->ContainsCallback(callback_wrapper->Id());
+    recordreplay::Assert("[RUN-1335-1336] IdleTaskFired A %d %d",
+                         callback_wrapper->Id(),
+                         hasCallback);
+
+    if (hasCallback)
```

Comment:
As before, can we gate this test and behavior change on whether we are recording/replaying and have some feature enabled, so we can isolate its effects if need be?  It is important to do this for non-trivial behavior changes we make to chromium, or behavior regressions will be very difficult to track down.

Reply:
Whoops... I forgot again. I need to start remembering this.